block-storage/src/main/scala/coop/rchain/blockstorage/deploy/LMDBDeployStorage.scala

```diff
@@ -0,0 +1,112 @@
+package coop.rchain.blockstorage.deploy
+
+import java.nio.ByteBuffer
+import java.nio.file.Path
+
+import cats.syntax.traverse._
+import cats.syntax.applicative._
+import cats.syntax.functor._
+import cats.syntax.flatMap._
+import cats.instances.list._
+import cats.effect.{Resource, Sync}
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.dag.codecs.codecSignedDeployData
+import coop.rchain.blockstorage.util.io.IOError.RaiseIOError
+import coop.rchain.blockstorage.util.io.{makeDirectory, notExists, IOError}
+import coop.rchain.casper.protocol.DeployData
+import coop.rchain.crypto.signatures.Signed
+import coop.rchain.lmdb.LMDBStore
+import coop.rchain.shared.ByteStringOps._
+import org.lmdbjava.DbiFlags.MDB_CREATE
+import org.lmdbjava.{Env, EnvFlags}
+import scodec.Attempt.{Failure, Successful}
+import scodec.DecodeResult
+import scodec.bits.BitVector
+
+import scala.collection.mutable.ArrayBuffer
+
+case object DeployEncodeError extends Exception
+case object DeployDecodeError extends Exception
+
+class LMDBDeployStorage[F[_]: Sync] private (storage: LMDBStore[F]) extends DeployStorage[F] {
+  def add(deploys: List[Signed[DeployData]]): F[Unit] =
+    deploys
+      .traverse(
+        d =>
+          codecSignedDeployData.encode(d) match {
+            case Successful(value) =>
+              val signature        = d.sig.toDirectByteBuffer
+              val serializedDeploy = ByteString.copyFrom(value.toByteArray).toDirectByteBuffer
+              (signature -> serializedDeploy).pure[F]
+            case Failure(_) =>
+              Sync[F].raiseError[(ByteBuffer, ByteBuffer)](DeployEncodeError)
```

Comment:
This will short-circuit, right? Is it the best thing to do?

Reply:
I can't think of anything you logically do if our scodec codec can't handle some types of deploys apart from crashing. I guess to make the system less brittle we could print to the node user "sorry, we couldn't serialize your deploy" without crashing.